In [47]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')

In [2]:
url = 'https://drive.google.com/uc?id=1N0kM_zi-iU3yvr6BEJf21aiF599PtA0U'
df = pd.read_csv(url)
df

,Car_id,Date,Customer Name,Gender,Annual Income,Dealer_Name,Company,Model,Engine,Transmission,Color,Price ($),Dealer_No,Body Style,Phone,Dealer_Region
0,C_CND_000001,1/2/2022,Geraldine,Male,13500,Buddy Storbeck's Diesel Service Inc,Ford,Expedition,Double Overhead Camshaft,Auto,Black,26000,06457-3834,SUV,8264678,Middletown
1,C_CND_000002,1/2/2022,Gia,Male,1480000,C & M Motors Inc,Dodge,Durango,Double Overhead Camshaft,Auto,Black,19000,60504-7114,SUV,6848189,Aurora
2,C_CND_000003,1/2/2022,Gianna,Male,1035000,Capitol KIA,Cadillac,Eldorado,Overhead Camshaft,Manual,Red,31500,38701-8047,Passenger,7298798,Greenville
3,C_CND_000004,1/2/2022,Giselle,Male,13500,Chrysler of Tri-Cities,Toyota,Celica,Overhead Camshaft,Manual,Pale White,14000,99301-3882,SUV,6257557,Pasco
4,C_CND_000005,1/2/2022,Grace,Male,1465000,Chrysler Plymouth,Acura,TL,Double Overhead Camshaft,Auto,Red,24500,53546-9427,Hatchback,7081483,Janesville
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
23901,C_CND_023902,12/31/2023,Martin,Male,13500,C & M Motors Inc,Plymouth,Voyager,Overhead Camshaft,Manual,Red,12000,60504-7114,Passenger,8583598,Pasco
23902,C_CND_023903,12/31/2023,Jimmy,Female,900000,Ryder Truck Rental and Leasing,Chevrolet,Prizm,Double Overhead Camshaft,Auto,Black,16000,06457-3834,Hardtop,7914229,Middletown
23903,C_CND_023904,12/31/2023,Emma,Male,705000,Chrysler of Tri-Cities,BMW,328i,Overhead Camshaft,Manual,Red,21000,99301-3882,Sedan,7659127,Scottsdale
23904,C_CND_023905,12/31/2023,Victoire,Male,13500,Chrysler Plymouth,Chevrolet,Metro,Double Overhead Camshaft,Auto,Black,31000,53546-9427,Passenger,6030764,Austin


In [3]:
#What is the average selling price of cars for each dealer, and how does it compare across different dealers
df.groupby("Dealer_Name",as_index=False)['Price ($)'].mean().sort_values(by='Price ($)')

,Dealer_Name,Price ($)
0,Buddy Storbeck's Diesel Service Inc,27217.261563
17,Pitre Buick-Pontiac-Gmc of Scottsdale,27404.248408
3,Chrysler Plymouth,27555.526400
12,McKinney Dodge Chrysler Jeep,27684.096979
6,Clay Johnson Auto Sales,27816.027113
14,Nebo Chevrolet,27818.889415
10,Hatfield Volkswagen,27853.712242
15,New Castle Ford Lincoln Mercury,27867.131955
18,Progressive Shippers Cooperative Association No,27884.264036
21,Ryder Truck Rental and Leasing,27914.988782


The dealers with the lowest average car prices are:

Buddy Storbeck's Diesel Service Inc — ₹27,217

Pitre Buick-Pontiac-Gmc of Scottsdale — ₹27,404

Chrysler Plymouth — ₹27,555

These dealers seem to offer more budget-friendly vehicles compared to others.

| Price Tier | Range             | Dealers                                        |
| ---------- | ----------------- | ---------------------------------------------- |
|   Low      | < ₹27,600         | Buddy Storbeck, Pitre Buick, Chrysler Plymouth |
|   Medium   | ₹27,600 – ₹28,200 | Majority of dealers                            |
|   High     | > ₹28,200         | Classic Chevy, Rabun Used Car Sales, U-Haul CO |

In [4]:
#Which car brand (Company) has the highest variation in prices, and what does this tell us about the pricing trends
df.groupby("Company",as_index=False)['Price ($)'].std().sort_values(ascending=False,by="Price ($)")

,Company,Price ($)
15,Lincoln,19658.050211
24,Saab,19653.740089
4,Cadillac,19517.120220
21,Plymouth,19065.997338
14,Lexus,17852.923492
3,Buick,17142.232626
17,Mercury,16445.172195
19,Nissan,16214.264017
25,Saturn,15990.223671
8,Ford,15849.090227


Licon shows the highest variation in prices(std ≈ 19,658) followed closely to saab and cadillac
this indicates thst these companies sell vechiles across a very wide spectrum from lower priced models to premium and luxary cars. The large spread suggest these brands cater to multiple customer segments and offer diverse model ranges with significance differences in specifications , features, and market positioning

on the otheer hand, Hundai shows the lowest price varitiion (std≈3,486),meaning most hundai cars falls with in a narrow consistant brice band. this consistency indicates a focused strategy toward budget or mid range segments with less variation among models

Overalll brands with higher price variation have a more uniform pricing structure.


In [5]:
#What is the distribution of car prices for each transmission type
#and how do the interquartile ranges compare
df.groupby("Transmission")['Price ($)'].describe()

,count,mean,std,min,25%,50%,75%,max
Transmission,,,,,,,,
Auto,12571.0,28248.525972,13747.070597,1200.0,19000.0,24000.0,35500.0,75700.0
Manual,11335.0,27914.710631,15862.871978,1700.0,17000.0,22001.0,34000.0,85800.0


In [6]:
iqr = df.groupby('Transmission')['Price ($)'].apply(lambda x: x.quantile(0.75) - x.quantile(0.25))
iqr


Transmission
Auto      16500.0
Manual    17000.0
Name: Price ($), dtype: float64

The distribution of car prices differs slightly between transmission types.
Automatic cars have prices mainly between $19,000 and $35,500, with a median price of $24,000, while manual cars range from $17,000 to $34,000, with a median of $22,001.

Comparing the interquartile ranges (IQR), manual cars have a slightly higher IQR ($17,000) compared to automatic cars ($16,500). This indicates that manual car prices are a bit more spread out in the middle 50%, while automatic car prices are slightly more consistent.

In [7]:
#What is the distribution of car prices across different regions

In [8]:
df.groupby("Dealer_Region")['Price ($)'].describe()

,count,mean,std,min,25%,50%,75%,max
Dealer_Region,,,,,,,,
Aurora,3130.0,28334.626837,15026.207252,9000.0,18001.0,23000.0,35000.0,85800.0
Austin,4135.0,28341.603628,14903.884549,9000.0,18001.0,23801.0,35001.0,85601.0
Greenville,3128.0,28180.819054,15101.538328,1200.0,18001.0,22500.0,34500.0,85200.0
Janesville,3821.0,27833.350955,14344.995638,4300.0,18001.0,23000.0,34000.0,85400.0
Middletown,3128.0,27856.338875,14619.842395,1700.0,18000.0,22750.0,34000.0,85300.0
Pasco,3131.0,28119.039923,14659.315941,9000.0,18500.5,23000.0,34000.0,85600.0
Scottsdale,3433.0,27954.958928,14902.916820,1450.0,18000.0,22600.0,33500.0,85001.0


In [11]:
#What is the distribution of cars based on body styles
df['Body Style'].value_counts()


Body Style
SUV          6374
Hatchback    6128
Sedan        4488
Passenger    3945
Hardtop      2971
Name: count, dtype: int64

In [18]:
#How does the average selling price of cars vary by customer gender and annual income
df.groupby('Gender')['Price ($)'].mean()


Gender
Female    28277.265270
Male      28039.429407
Name: Price ($), dtype: float64

In [43]:
print(df['Annual Income'].max())
print(df['Annual Income'].median())
print(df['Annual Income'].min())

11200000
735000.0
10080


In [39]:
bins = [0, 500000, 1000000, 2000000, 4000000, 8000000, float('inf')]
labels = ['0–5L', '5–10L', '10–20L', '20–40L', '40–80L', '80L+']

df['Income Bracket'] = pd.cut(df['Annual Income'], bins=bins, labels=labels)

df.groupby('Income Bracket',observed=True)['Price ($)'].mean()


Income Bracket
0–5L      28086.588267
5–10L     28011.726423
10–20L    28016.841147
20–40L    28939.841727
40–80L    29428.955056
80L+      26001.000000
Name: Price ($), dtype: float64

In [27]:
#What is the distribution of car prices by region, and how does the number of cars sold vary by region
df.groupby('Dealer_Region')['Price ($)'].describe()


,count,mean,std,min,25%,50%,75%,max
Dealer_Region,,,,,,,,
Aurora,3130.0,28334.626837,15026.207252,9000.0,18001.0,23000.0,35000.0,85800.0
Austin,4135.0,28341.603628,14903.884549,9000.0,18001.0,23801.0,35001.0,85601.0
Greenville,3128.0,28180.819054,15101.538328,1200.0,18001.0,22500.0,34500.0,85200.0
Janesville,3821.0,27833.350955,14344.995638,4300.0,18001.0,23000.0,34000.0,85400.0
Middletown,3128.0,27856.338875,14619.842395,1700.0,18000.0,22750.0,34000.0,85300.0
Pasco,3131.0,28119.039923,14659.315941,9000.0,18500.5,23000.0,34000.0,85600.0
Scottsdale,3433.0,27954.958928,14902.916820,1450.0,18000.0,22600.0,33500.0,85001.0


In [30]:
df.Dealer_Region.value_counts()

Dealer_Region
Austin        4135
Janesville    3821
Scottsdale    3433
Pasco         3131
Aurora        3130
Greenville    3128
Middletown    3128
Name: count, dtype: int64

1. Number of Cars Sold by Region

The number of cars varies noticeably across regions:

Austin has the highest number of cars (4,135)

Followed by Janesville (3,821)

Scottsdale, Pasco, Aurora, Greenville, and Middletown each have around 3,100–3,400 cars

Overall, Austin shows the highest market activity, while the other regions have similar sales volumes.

2. Distribution of Car Prices by Region

Across all regions, car prices have a similar pattern:

Median prices are close (around 22,500 – 23,801)

Q1 values stay near 18,000

Q3 values stay near 34,000 – 35,001

Maximum prices in all regions reach 85,000+, showing availability of luxury cars

Minimum prices range from 1,200 to 9,000, indicating availability of budget cars too

Standard deviation (≈14k–15k) is high in all regions, indicating large price variability

In [34]:
#How does the average car price differ between cars with different engine 
df.groupby('Engine')['Price ($)'].mean().sort_values()

Engine
Overhead Camshaft            27914.710631
Double Overhead Camshaft    28248.525972
Name: Price ($), dtype: float64

In [ ]:
#DOHC cars are costlier
#OHC cars are slightly cheaper

In [48]:
df.groupby('Income Bracket')['Price ($)'].mean()

Income Bracket
0–5L      28086.588267
5–10L     28011.726423
10–20L    28016.841147
20–40L    28939.841727
40–80L    29428.955056
80L+      26001.000000
Name: Price ($), dtype: float64

The average car price varies across different annual income brackets, but the trend is not strictly linear.
Customers in the lower to mid-income brackets (0–5L, 5–10L, and 10–20L) purchase cars with very similar average prices of around ₹28,000, indicating a consistent preference for mid-range vehicles across these groups.

Customers in the higher income brackets (20–40L and 40–80L) purchase slightly more expensive vehicles, with average prices rising to ₹28,940 and ₹29,429, showing a modest increase in spending power.

Interestingly, the highest income bracket (80L+) shows a lower average price (₹26,001), likely due to very few data points or outliers in this category.

Overall, the data suggests that while income does influence car price slightly, most customers, regardless of income, purchase cars within a similar mid-range price segment.

In [52]:
#What are the top 5 car models with the highest number of sales, and how does their price distribution look
top5=df.Model.value_counts().head()
top5

Model
Diamante      418
Prizm         411
Silhouette    411
Passat        391
Ram Pickup    383
Name: count, dtype: int64

In [56]:
top5_models = df[df.Model.isin(top5.index)]
t5_dist = top5_models.groupby('Model')['Price ($)'].describe()
t5_dist

,count,mean,std,min,25%,50%,75%,max
Model,,,,,,,,
Diamante,418.0,22268.303828,9770.768958,12000.0,18000.0,19001.0,22000.0,49801.0
Passat,391.0,19600.728900,8218.521683,9000.0,12001.0,20000.0,27001.0,32501.0
Prizm,411.0,22838.457421,8012.531045,15000.0,16000.0,22000.0,25000.0,44700.0
Ram Pickup,383.0,27820.864230,13297.609534,16000.0,19000.0,21000.0,39300.0,57750.0
Silhouette,411.0,29858.513382,17940.571762,12000.0,14001.0,25000.0,34500.0,60750.0


In [59]:
# How does car price vary with engine size across different car colors, and which colors have the highest price variation
df.groupby(['Engine','Color'])['Price ($)'].mean()

Engine                     Color     
Double Overhead Camshaft  Black         25766.451103
                           Pale White    30347.987460
                           Red           27434.392314
Overhead Camshaft          Black         31732.967164
                           Pale White    24327.767694
                           Red           30153.405229
Name: Price ($), dtype: float64

In [63]:
df.groupby('Color')['Price ($)'].std()

Color
Black         15286.065976
Pale White    14077.346859
Red           15519.360962
Name: Price ($), dtype: float64

The average selling price of cars varies across different combinations of engine type and color. Among Double Overhead Camshaft (DOHC) cars, Pale White models have the highest average price (₹30,348), followed by Red and Black variants. For Overhead Camshaft (OHC) engines, Black cars show the highest mean price (₹31,733), while Pale White cars show the lowest (₹24,328).

This indicates that both engine type and color together influence car pricing, with certain color–engine combinations (such as OHC Black) positioned in higher price segments.

When analyzing price variation using standard deviation, Red cars exhibit the highest price variability (std ≈ 15,519), followed closely by Black cars. Pale White cars have the lowest variation, suggesting more consistent pricing within that color category.

Overall, Red-colored vehicles show the widest price spread, while Pale White vehicles are priced more uniformly.

In [73]:
#Is there any seasonal trend in car sales based on the date of sale
df['Date'] = pd.to_datetime(df['Date'])
df['month'] = df['Date'].dt.month
df['month'].value_counts().sort_index()

month
1      790
2      735
3     1535
4     1655
5     1895
6     1715
7     1725
8     1705
9     3305
10    1830
11    3470
12    3546
Name: count, dtype: int64

A clear seasonal trend is observed in the monthly car sales data. Sales are lowest in January (790) and February (735), indicating a slow start to the year. From March to August, sales gradually increase and remain relatively stable, ranging between 1500 and 1900 units.

A sharp rise in sales is seen during September (3305), followed by even higher peaks in November (3470) and December (3546), which are the highest sales months of the year.

This pattern suggests a strong seasonal effect, with significantly higher car purchases occurring during the festival season and year-end period, likely driven by festive discounts, promotions, and consumer buying behavior. Overall, the data shows a clear upward seasonal trend in the last quarter of the year.

In [74]:
# How does the car price distribution change when considering different combinations 
#of body style and transmission type
df.groupby(['Body Style', 'Transmission'])['Price ($)'].describe()

count          mean           std      min      25%  \
Body Style Transmission                                                         
Hardtop    Auto          1408.0  31520.188210  13687.756740  12000.0  21001.0   
           Manual        1563.0  27016.943698  14072.501848  11000.0  17575.5   
Hatchback  Auto          3458.0  25910.544824  13144.014337   4200.0  18001.0   
           Manual        2670.0  28702.550562  16284.049616   1700.0  17000.0   
Passenger  Auto          2020.0  28915.835149  13946.481932   9000.0  19000.0   
           Manual        1925.0  28969.521039  18523.684163  10000.0  16001.0   
SUV        Auto          3086.0  27501.404407  13299.927913   1200.0  19000.0   
           Manual        3288.0  26079.019161  14621.526541   2200.0  17000.0   
Sedan      Auto          2599.0  29955.294344  14310.910860  12000.0  19000.0   
           Manual        1889.0  29664.271572  15440.056434  11000.0  19001.0   

                             50%      75%      max  
Body Style Transmission                             
Hardtop    Auto          27000.0  43125.0  75501.0  
           Manual        22001.0  29000.0  82500.0  
Hatchback  Auto          21500.0  31000.0  75700.0  
           Manual        23000.0  36001.0  85600.0  
Passenger  Auto          25000.0  38000.0  69801.0  
           Manual        22000.0  37000.0  85800.0  
SUV        Auto          25000.0  33000.0  75501.0  
           Manual        22000.0  28000.0  85500.0  
Sedan      Auto          26000.0  39500.0  69980.0  
           Manual        25000.0  36600.0  69700.0

The car price distribution changes noticeably across different body style and transmission combinations. Automatic cars are generally more expensive than manual ones in most body styles (Hardtop, SUV, Sedan).
However, manual Hatchbacks have higher average prices and the widest price variation, showing many mixed trims.

Overall:

Hardtop Auto → highest prices

Hatchback Manual → highest variation

Sedan & Passenger → small difference between Auto and Manual

Manual variants usually show higher spread (std), meaning more diverse pricing

Body style affects price more strongly than transmission type

In [76]:
#How does the average car price vary across different car models and engine types?
df.groupby('Model')['Price ($)'].mean().sort_values(ascending=False)

Model
Catera     56836.933333
DeVille    46154.050000
Contour    45309.079646
Integra    44604.000000
Tacoma     44576.743017
               ...     
Montero    16035.000000
C-Class    14538.414474
Escort     12901.092105
Cirrus     12727.243243
Mirage      9063.368421
Name: Price ($), Length: 154, dtype: float64

In [77]:
df.groupby('Engine')['Price ($)'].mean()

Engine
Double Overhead Camshaft    28248.525972
Overhead Camshaft            27914.710631
Name: Price ($), dtype: float64

In [62]:
df.columns

Index(['Car_id', 'Date', 'Customer Name', 'Gender', 'Annual Income',
       'Dealer_Name', 'Company', 'Model', 'Engine', 'Transmission', 'Color',
       'Price ($)', 'Dealer_No ', 'Body Style', 'Phone', 'Dealer_Region',
       'Income Bracket'],
      dtype='object')